### PR TITLE
add cli mode to run nameifier as a standalone cli app

### PR DESCRIPTION
**Pull request by danpilch**
add cli mode and cli args with `kong` to run nameifier as a standalone cli app

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -65,24 +67,32 @@ func loggingMiddleware(next http.Handler) http.Handler {
...
+		namer := NewNameifier()
+		for i := 0; i < Config.Count; i++ {
+			name, err := namer.Nameify(fmt.Sprintf("%s-%d", Config.SeedString, i))
+			if err != nil {
+				fmt.Printf("Failed to nameify! %s", err)
+				return
+			}
+			allNames = append(allNames, name)
+		}
+		fmt.Println(strings.Join(allNames, "\n"))
```
Comment (danpilch) [inline main.go]: maybe this should be in a `func`

File: main.go [hunk 2]
```diff
@@ -70,30 +72,38 @@ func loggingMiddleware(next http.Handler) http.Handler {
...
-	router := mux.NewRouter()
-	router.Use(loggingMiddleware)
+	kong.Parse(&Config)
 
-	router.HandleFunc("/nameifier/{seed}/{count}", nameHandler).Methods("GET")
-	router.HandleFunc("/nameifier/{seed}", blankHandler).Methods("GET")
-	router.HandleFunc("/nameifier/", blankHandler).Methods("GET")
-	router.HandleFunc("/nameifier/", blankHandler).Methods("GET")
-	router.PathPrefix("/").Handler(uiFs)
+	// By default start server
```
Comment (relistan) [inline main.go:81]: We should explicitly add a `--cli` flag or similar that we switch on so there's not any magic.

File: main.go [hunk 3]
```diff
@@ -70,30 +72,38 @@ func loggingMiddleware(next http.Handler) http.Handler {
...
+	if Config.SeedString == "" {
```
Comment (relistan) [inline main.go]: I'd leave this the way it was and then put `if Config.Cli { runCli(config); return }` or similar (formatted properly) to shortcut out of the code.

Comment (danpilch) [reply main.go]: Okay I'll work on that :)

File: main.go [hunk 4]
```diff
@@ -70,30 +72,38 @@ func loggingMiddleware(next http.Handler) http.Handler {
...
+		var allNames []string
+		namer := NewNameifier()
+		for i := 0; i < Config.Count; i++ {
+			name, err := namer.Nameify(fmt.Sprintf("%s-%d", Config.SeedString, i))
+			if err != nil {
+				fmt.Printf("Failed to nameify! %s", err)
+				return
+			}
+			allNames = append(allNames, name)
+		}
```
Comment (relistan) [inline main.go]: This block is mostly shared with the http service. Let's break it out into a function and call it from both.

**Review by relistan — APPROVED** [review]
Some code changes I'd like to see, but looking good for functionality

**Review by relistan — APPROVED** [review]
Great! Thanks

**Pull request merged**